src/uu/dd/src/progress.rs

```diff
@@ -154,20 +154,23 @@ impl ProgUpdate {
         // If the number of bytes written is sufficiently large, then
         // print a more concise representation of the number, like
         // "1.2 kB" and "1.0 KiB".
-        if btotal < 1000 {
-            write!(
+        match btotal {
+            1 => write!(
+                w,
+                "{}{} byte copied, {:.1} s, {}/s{}",
+                carriage_return, btotal, duration, transfer_rate, newline,
+            ),
+            btotal if btotal < 1000 => write!(
```

Comment:
You could write this as
```rust
match btotal {
    1 => {}
    0..=999 => {}
    0..=1023 => {}
    _ => {}
}
```
Sadly, we can't use `..1000` because inclusive and half-open patterns are unstable.

I'll leave it up to you whether you think that's clearer or not.

Reply:
Interesting, let's go with that.